### PR TITLE
[Enhancement] support creating lake statistic tables

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Because statistics tables are built on local. If there are only lake tables in a cluster, after you put all BE down and get some new BE up, the statistics tables will not exist and the query plan would change. So this pr fix this bug.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by abc982627271** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 18 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|12|18|66.67%|[124, 126, 169, 171, 213, 215]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 14 / 22 (63.64%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|1|2|50.00%|[110]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|13|20|65.00%|[83, 127, 129, 172, 174, 216, 218]|

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsMetaManager.java [hunk 1]
```diff
@@ -118,16 +121,32 @@ private boolean createSampleStatisticsTable() {
...
-                properties,
-                null,
-                "");
+        // if use_staros, create lake table
+        CreateTableStmt stmt = null;
+        if (Config.use_staros) {
+            stmt = new CreateTableStmt(false, false,
+                    tableName,
+                    StatisticUtils.buildStatsColumnDef(StatsConstants.SAMPLE_STATISTICS_TABLE_NAME),
+                    "olap",
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsMetaManager.java:130]:
```suggestion
                    "starrocks",
```

Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsMetaManager.java:130]: Better to use `com.starrocks.sql.ast.CreateTableStmt.LAKE_ENGINE_NAME`

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsMetaManager.java [hunk 2]
```diff
@@ -118,16 +121,32 @@ private boolean createSampleStatisticsTable() {
...
+                    new KeysDesc(KeysType.UNIQUE_KEYS, KEY_COLUMN_NAMES),
+                    null,
+                    new HashDistributionDesc(10, KEY_COLUMN_NAMES),
+                    properties,
+                    null,
+                    "");
+        }
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsMetaManager.java:148]: string engine = Config.use_staros ? LAKER : OLAP;
stmt = new Create.....

**Comment by abc982627271** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsMetaManager.java [hunk 3]
```diff
@@ -176,11 +185,14 @@ private boolean createHistogramStatisticsTable() {
...
+        // if use_staros, create lake table
+        CreateTableStmt stmt = null;
+        String engine = Config.use_staros ? CreateTableStmt.LAKE_ENGINE_NAME : "olap";
+        stmt = new CreateTableStmt(false, false,
                 tableName,
                 StatisticUtils.buildStatsColumnDef(StatsConstants.HISTOGRAM_STATISTICS_TABLE_NAME),
-                "olap",
-                new KeysDesc(KeysType.PRIMARY_KEYS, HISTOGRAM_KEY_COLUMNS),
+                engine,
+                new KeysDesc(KeysType.UNIQUE_KEYS, HISTOGRAM_KEY_COLUMNS),
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsMetaManager.java:195]: When `use_staros` is false, the primary key table should be used.

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsMetaManager.java [hunk 4]
```diff
@@ -118,10 +121,13 @@ private boolean createSampleStatisticsTable() {
         Map<String, String> properties = Maps.newHashMap();
         int defaultReplicationNum = Math.min(3, GlobalStateMgr.getCurrentSystemInfo().getTotalBackendNumber());
         properties.put(PropertyAnalyzer.PROPERTIES_REPLICATION_NUM, Integer.toString(defaultReplicationNum));
-        CreateTableStmt stmt = new CreateTableStmt(false, false,
+        // if use_staros, create lake table
+        CreateTableStmt stmt = null;
+        String engine = Config.use_staros ? CreateTableStmt.LAKE_ENGINE_NAME : "olap";
+        stmt = new CreateTableStmt(false, false,
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsMetaManager.java:127]:
```suggestion
        String engine = Config.use_staros ? CreateTableStmt.LAKE_ENGINE_NAME : "olap";
        CreateTableStmt stmt = new CreateTableStmt(false, false,
```

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 10 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|1|2|50.00%|[110]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|7|8|87.50%|[83]|

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by abc982627271** [comment]
run starrocks_admit_test

**Pull request merged**